MerlinAU.sh

```diff
@@ -211,12 +211,40 @@ Toggle_LEDs_PID=""
 FW_UpdateCheckState="TBD"
 FW_UpdateCheckScript="/usr/sbin/webs_update.sh"
 
-##---------------------------------------##
-## Added by ExtremeFiretop [2024-Jan-21] ##
-##---------------------------------------##
+
+##--------------------------------------##
+## Added by Martinski W. [22023-Nov-24] ##
+##--------------------------------------##
+#---------------------------------------------------------#
+# The USB-attached drives can have multiple partitions
+# with different file systems (NTFS, ext3, ext4, etc.),
+# which means that multiple mount points can be found.
+# So for the purpose of choosing a default value here
+# we will simply select the first mount point found.
+# Users can later on change it by typing a different
+# mount point path or directory using the Main Menu.
+#---------------------------------------------------------#
+_GetDefaultUSBMountPoint_()
+{
+   local mounPointPath  retCode=0
+   local mountPointRegExp="/dev/sd.* /tmp/mnt/.*"
+
+   mounPointPath="$(grep -m1 "$mountPointRegExp" /proc/mounts | awk -F ' ' '{print $2}')"
+   [ -z "$mounPointPath" ] && retCode=1
+   echo "$mounPointPath" ; return "$retCode"
+}
+
+##------------------------------------------##
+## Modified by ExtremeFiretop [2024-Jan-26] ##
+##------------------------------------------##
 _migrate_settings_() {
+    local USBMountPoint="$(_GetDefaultUSBMountPoint_)"
     local old_settings_dir="${ADDONS_PATH}/${ScriptFNameTag}"
     local new_settings_dir="${ADDONS_PATH}/${ScriptFNameTag}.d"
```

Comment:
This doesn't look correct. The default mount point is not necessarily the one selected by the user when setting it up via the menu, especially if more than one USB-attached drive exists or multiple partitions are found.